shipment/buyer/receive/download-document.md

```diff
@@ -0,0 +1,137 @@
+---
+description: How to download a document from a shipment
+---
+
+# Download a document attached to a shipment
+
+You can download documents using two methods:
+
+1. If the webhook shipment `document` contains an `objectId`: Download the supplier's document from the Tradecloud object-storage using the `objectId` and `downloadUrl`.
+2. If the webhook shipment `document` contains an `url`: Download the document from the supplier's content server.
+
+{% hint style="info" %}
+If both `objectId` and `url` are missing in the `document`, no file is available for download.
+{% endhint %}
+
+## Method 1. Download the document from the Tradecloud object-storage
+
+### Step 1. Retrieve the `objectId` from the document
+
+Please check the `documents` fields in:
+
+```text
+"documents": [
+  {
+    "code": "10007563",
+    "revision": "B",
+    "name": "Tradecloud API Manual",
+    "description": "General document",
+    "type": "General",
+    "objectId": "05d251f0-0166-4e1e-8110-8d7a507b18f4",
+    "meta": {
+      "lastUpdatedAt": "2019-12-31T10:11:12"
+    }
+  }
+],
+```
+
+In this example the `objectId` is:
+
+```text
+05d251f0-0166-4e1e-8110-8d7a507b18f4
+```
+
+### Step. 2. Retrieve the `downloadUrl` from the object storage
+
+{% api-method method="get" host="https://api.accp.tradecloud1.com/v2" path="/object-storage/document/{objectId}/metadata" %}
+{% api-method-summary %}
+Get document metadata
+{% endapi-method-summary %}
+
+{% api-method-description %}
+
+{% endapi-method-description %}
+
+{% api-method-spec %}
+{% api-method-request %}
+{% api-method-path-parameters %}
+{% api-method-parameter name="objectId" type="string" required=false %}
+document objectId
+{% endapi-method-parameter %}
+{% endapi-method-path-parameters %}
+
+{% api-method-headers %}
+{% api-method-parameter name="Authorization" type="string" required=false %}
+Bearer Access-Token
+{% endapi-method-parameter %}
+{% endapi-method-headers %}
+{% endapi-method-request %}
+
+{% api-method-response %}
+{% api-method-response-example httpCode=200 %}
+{% api-method-response-example-description %}
+
+{% endapi-method-response-example-description %}
+
+```
+{
+  "id": "05d251f0-0166-4e1e-8110-8d7a507b18f4",
+  "filename": "test.pdf",
+  "contentType": "application/octet-stream",
+  "downloadUrl": "https://tradecloud-accp-documents.s3.eu-central-1.amazonaws.com/67aa8ece-5d41-496f-a94c-483e360b833b?response-content-disposition=attachment%3B%20filename%3Dtest.pdf&X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Date=20200406T203728Z&X-Amz-SignedHeaders=host&X-Amz-Expires=59&X-Amz-Credential=AKIAUXFSTTRHS7FN2JKY%2F20200406%2Feu-central-1%2Fs3%2Faws4_request&X-Amz-Signature=8bda9e2b4d810e0d7650e363373874c4f2e52aa53f1e46d82c5bf3eaf907832d"
```

Comment:
Maybe truncate this url for easier reading and security reasons (since this page is public, we don't have to show where our documents are shotred (eu-central-1) etc..

Reply:
I replaced the actual `downloadUrl` with a placeholder.